compiler/ir/ir.tree/src/org/jetbrains/kotlin/ir/util/ExternalDependenciesGenerator.kt

```diff
@@ -42,9 +44,9 @@ class ExternalDependenciesGenerator(val symbolTable: SymbolTable, private val ir
                 if (!symbol.isBound) {
                     irProviders.getDeclaration(symbol)
                 }
-                assert(symbol.isBound) { "$symbol unbound even after deserialization attempt" }
+                assert(symbol.isBound) { "${symbol.descriptor} unbound even after deserialization attempt" }
             }
-        } while (unbound.isNotEmpty())
+        } while (haveNotStabilized(unbound, symbolTable.allUnbound))
```

Comment:
The `haveNotStabilized` method seems very expensive computationally. I think all you have to do is to track in the do-while loop whether at least one previously-unbound symbol was bound, and the condition would be `while (unbound.isNotEmpty() || newSymbolBound)`

Reply:
That sounds better indeed, thanks! Addressed at d6ec9fc40a5722f81791aec67227c5630950e09c (and intentionally being a separate commit)